### DISCUSSION TITLE
unrecognized selector sent to instance using cordova plugin in capacitor

### DISCUSSION THREAD

**Discussion by xulihang** [ionic-team/capacitor#5595] [Q&A] [answered]
I am trying to use the cordova plugin in a capacitor project: https://github.com/xulihang/cordova-plugin-dynamsoft-barcode-reader , with [this wrapper](https://github.com/xulihang/awesome-cordova-plugins/tree/master/src/%40awesome-cordova-plugins/plugins/dynamsoft-barcode-scanner).

It works fine on Android. But I met the `unrecognized selector sent to instance` problem on iOS.

I've tried adding the linker flags according to [this issue](https://github.com/ionic-team/capacitor/issues/950), but it didn't solve the problem.

Any help would be appreciated.

The test project: https://github.com/xulihang/Ionic-React-Cordova-Barcode-Scanner

**Comment by xulihang** [comment, marked as answer]
Turns out this is a problem of the native library.

**Reply by DennisRein** [reply]
Did you just accept your faith or were you able to find a solution? I am facing the same issue currently
